### ISSUE TITLE
How to find the number of iterations while using tf.compat.v1.train.AdamOptimizer?

### ISSUE THREAD

**Issue by tusharpoddar** [tensorflow/tensorflow#41402]
I am actually. using the VGGish model to train my data and I am using the tf.compat.v1.train.AdamOptimizer to minimize my error. Is there a way in which I can print the number of iterations that the model does before printing the loss. Below is the code that I am using to train the model.

My definition of the optimizer -

```
optimizer = tf.train.AdamOptimizer(
            # time steps
            learning_rate=vggish_params.LEARNING_RATE,
            # tolerance level 
            epsilon=vggish_params.ADAM_EPSILON)
        optimizer.minimize(loss, global_step=global_step, name='train_op')
```

Loop where I am feeding the data each time. I want to print the number of iterations for each time I feed the data.

```
for _ in range(FLAGS.num_batches):
      #for normal batch training
      (features, labels) = _get_examples_batch()
      [num_steps, loss, _] = sess.run(
        # here the first array that we are passing in is called fetches
        # it is a list of graph elements
          [global_step_tensor, loss_tensor, train_op],
          # one has to be of the form of a tensor and the other in the form of a numpy array
          feed_dict={features_tensor: features, labels_tensor: labels})
      print('Step %d: loss %g' % (num_steps, loss))
```

**Comment by ravikyram** [comment]
@tusharpoddar 

Please, fill [issue template](https://github.com/tensorflow/tensorflow/issues/new/choose).

Request you to share colab link or simple standalone code with supporting files to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by tusharpoddar** [comment]
This is the python file that I am using to train my model. It has the definition of the model. It uses Adam optimizer. All I want to do is that each time when it prints the loss, I want to print the number of iterations that the model does before printing the loss. 


[vggish_train_demo.py.zip](https://github.com/tensorflow/tensorflow/files/4930310/vggish_train_demo.py.zip)

**Comment by ravikyram** [comment]
@tusharpoddar 

I think this is more related to models repo.Please, raise an issue in models repo by filling issue template from [here](https://github.com/tensorflow/models/issues/new/choose).Thanks!
